pkg/egressgateway/manager.go

```diff
@@ -39,21 +48,44 @@ type Manager struct {
 
 	// epDataStore stores endpointId to endpoint metadata mapping
 	epDataStore map[endpointID]*endpointMetadata
+
+	// identityAllocator is used to fetch identity labels for endpoint updates
+	identityAllocator identityCache.IdentityAllocator
 }
 
 // NewEgressGatewayManager returns a new Egress Gateway Manager.
-func NewEgressGatewayManager(k8sCacheSyncedChecker k8sCacheSyncedChecker) *Manager {
+func NewEgressGatewayManager(k8sCacheSyncedChecker k8sCacheSyncedChecker, identityAlocator identityCache.IdentityAllocator) *Manager {
 	manager := &Manager{
 		k8sCacheSyncedChecker: k8sCacheSyncedChecker,
 		policyConfigs:         make(map[policyID]*PolicyConfig),
 		epDataStore:           make(map[endpointID]*endpointMetadata),
+		identityAllocator:     identityAlocator,
 	}
 
 	manager.runReconciliationAfterK8sSync()
 
 	return manager
 }
 
+// getIdentityLabels waits for the global identities to be populated to the cache,
+// then looks up identity by ID from the cached identity allocator and return its labels.
+func (manager *Manager) getIdentityLabels(securityIdentity uint32) (labels.Labels, error) {
+	identityCtx, cancel := context.WithTimeout(context.Background(), option.Config.KVstoreConnectivityTimeout)
+	defer cancel()
+
+	waitForIdentityCacheOnce.Do(func() {
+		if err := manager.identityAllocator.WaitForInitialGlobalIdentities(identityCtx); err != nil {
+			log.WithError(err).Error("Failed to wait for initial global identities.")
```

Comment:
Unfortunately with this approach we are going to ignore any error returned by `WaitForInitialGlobalIdentities()` :disappointed: 

I checked internally and in the end it's fine to keep calling `WaitForInitialGlobalIdentities()` each time, so we should go back to the original approach and just get rid of the `waitForIdentityCacheOnce.Do()` (apologies for the back and forth on this change :pray:)

Reply:
Yeah I see what you meant.
NP, this is reverted back to 1st version.